assets/js/interactivity/hooks.tsx

```diff
@@ -0,0 +1,261 @@
+// @ts-nocheck
+
+/**
+ * External dependencies
+ */
+import { h, options, createContext, cloneElement } from 'preact';
+import { useRef, useCallback, useContext } from 'preact/hooks';
+/**
+ * Internal dependencies
+ */
+import { stores } from './store';
+import { nsPathParser } from './vdom';
+
+/** @typedef {import('preact').VNode} VNode */
+/** @typedef {typeof context} Context */
+/** @typedef {ReturnType<typeof getEvaluate>} Evaluate */
+
+/**
+ * @typedef {Object} DirectiveCallbackParams Callback parameters.
+ * @property {Object}   directives Object map with the defined directives of the element being evaluated.
+ * @property {Object}   props      Props present in the current element.
+ * @property {VNode}    element    Virtual node representing the original element.
+ * @property {Context}  context    The inherited context.
+ * @property {Evaluate} evaluate   Function that resolves a given path to a value either in the store or the context.
+ */
+
+/**
+ * @callback DirectiveCallback Callback that runs the directive logic.
+ * @param {DirectiveCallbackParams} params Callback parameters.
+ */
+
+/**
+ * @typedef DirectiveOptions Options object.
+ * @property {number} [priority=10] Value that specifies the priority to
+ *                                  evaluate directives of this type. Lower
+ *                                  numbers correspond with earlier execution.
+ *                                  Default is `10`.
+ */
+
+// Main context.
+const context = createContext( {} );
+
+let currentScope = null;
+let currentNamespace = null;
+
+export const getContext = < T extends object >(
+	namespace: string = currentNamespace
+): T => {
+	return currentScope.context[ namespace ];
+};
+
+export const getElementRef = () => currentScope.ref.current;
+
+export const getScope = () => currentScope;
+export const setScope = ( scope ) => {
+	currentScope = scope;
+};
+export const resetScope = () => {
+	currentScope = null;
+};
+
+export const setNamespace = ( namespace ) => {
+	currentNamespace = namespace;
+};
+export const resetNamespace = () => {
+	currentNamespace = null;
+};
```

Comment:
One of the next steps should be to turn these into stacks.

Reply:
Done in 96631ce688118e399f249826a568e26819e53f74.